.github/workflows/macos_tests.yaml

```
@@ -0,0 +1,126 @@
+name: MacOS Tests
+on:
+  push:
+    branches:
+      - main
+      - 'releases/**'
+  pull_request:
+    branches:
+      - main
+      - 'releases/**'
+
+# Restrict tests to the most recent commit.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_test_app:
+    name: Build Test Client
+    runs-on: macos-latest
+    outputs:
+      matrix: ${{ steps.testGen.outputs.tests }}
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: Install dependencies
+        shell: bash
+        run: |
+          pip3 install -r requirements.txt
+          brew install ninja
+
+      - name: Install Qt6
+        shell: bash
+        run: |
+          wget https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mozillavpn.v2.mozillavpn.cache.level-3.toolchains.v3.qt-mac.latest/artifacts/public%2Fbuild%2Fqt6_mac.zip -O qt6_mac.zip
+          unzip -a -d ${{github.workspace}} qt6_mac.zip
```

Super nit, we seem to go back and forth between the 2 styles of formatting. I suggested we add a space between all `{{` for the sake of consistency but there are also plenty in the existing old code without any spaces so if you don't want to make this change, I'm totally fine with it as is. 

```suggestion
          unzip -a -d ${{ github.workspace }} qt6_mac.zip
```